### PR TITLE
Feature/porting guide edits

### PR DESCRIPTION
**Pull request by DellaBitta**
[ Description ]
Revised Porting Guide with updates to the build process and to make it more concise.

- Updated the version of the porting guide to fall in line with the formatting of source paths of our upcoming CC3200 Tutorial.
- Removed section about Python Client, instead directing the reader to the xiPy Repository.
- Removed references of the xi_client_c path that remained since before we switched repos.
- Added full paths to files where they would have been helpful
- Condensed sections to make the document more concise.
- Added a new Further Reading section to the bottom of the document.
- Updated TLS config flag to use the new tls_bsp nomenclature. 

[ Reviewer ]
@atigyi 
@olgierdh 

[ QA ]
Ensured anchor links work

[ Release Notes ]
Revised Porting Guide to make it more clear and concise.

### CODE REVIEW

File: doc/porting_guide.md [hunk 1]
```diff
@@ -3,268 +3,284 @@
...
+An event system is included in the client which can be used by your client application as well. The client's event system is a platform independant implementation that doesn't require porting to any platform-specific timers or tasks.
 
 The client provides this functionality while remaining small enough to reside on low resource MCUs, consumes minimal memory, processor power and energy consumption.
 
 The codebase is built against C99 standards.
 
-For more information place see the Xively C Client user guide in the Xively Developer Center.
+For more information please see the Xively C Client User Guide: `doc/user_guide.md`.
 
-#### Python Client
```
Comment (atigyi) [inline doc/porting_guide.md]: bye-bye

File: doc/porting_guide.md [hunk 2]
```diff
@@ -3,268 +3,284 @@
...
-#### C Client
+## C Client Overview
 
-The C Client is designed for low resources platforms. The single threaded implementation is non-blocking, which allows it to operate on NoOs platforms. It also unlocks the ability for processing outgoing and incoming messages concurrently.
+The Xively C Client is designed for low resources platforms. The single threaded implementation is non-blocking, which allows it to operate on NoOs platforms. The implementation also supports seamless concurrent processing of outgoing and incoming messages.
 
-An event system is included and can be used by client applications to schedule publications at regular intervals without any platform-specific timers or tasks.
+An event system is included in the client which can be used by your client application as well. The client's event system is a platform independant implementation that doesn't require porting to any platform-specific timers or tasks.
 
 The client provides this functionality while remaining small enough to reside on low resource MCUs, consumes minimal memory, processor power and energy consumption.
```
Comment (atigyi) [inline doc/porting_guide.md]: I know you haven't touched this sentence but "consume" appears twice. Could you rephrase it?

File: doc/porting_guide.md [hunk 3]
```diff
@@ -3,268 +3,284 @@
...
+The Board Support Package (BSP) is the well-defined set of functions that the Xively Client invokes to interact with a platform's specific networking, [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security), memory, random number generator, and time implementations. 
+
+We've organized the sources of the Xivey Client so that your BSP implementation is in just a few files that reside in the directory (`src/bsp`).  This is where you should focus your attention when writing porting the Xively Client to your device SDK.
 
 The BSP was designed to minimize the time it takes to make a port of the library. With it an engineer can ignore the MQTT codec or the non-blocking / asynchronous engine that resides in the main library source.
 
 All of the BSP **function declarations** can be found under the _xi\_client\_c/include/bsp_
```
Comment (atigyi) [inline doc/porting_guide.md]: sorry xi_client_c still resides here. because of my escape characters. You can remove those escape chars, it is only needed by my local chrome-md-plugin. sorry.

File: doc/porting_guide.md [hunk 4]
```diff
@@ -3,268 +3,284 @@
...
-them to the actual *application*. Obviously these steps need to be tailored to the target platform.
+Building a Xively C Client consists of building the Xively C Client *library*, the TLS library and linking them to the actual *application*. Obviously these steps need to be tailored to the target platform.
 
-### Preceding Step - A Native Build
+### Suggested First Step - A Native Build
 
-We recommend building the Xively Client on OSX or Linux/Unix before attempting to cross compile the sources to another target platform. This process could serve as an introduction to the basic architecture, tools and steps of building the Xively Client. This basic knowledge will be handy later when doing a full cross-compiled port of the software to an embedded device later.
+We recommend building the Xively Client on OSX or Linux/Unix before attempting to cross compile the sources to another target platform. This process could serve as an introduction to the basic architecture, tools and building stepos of the Xively Client source. This basic knowledge will be handy later when doing a full cross-compiled port of the software to an embedded device.
```
Comment (atigyi) [inline doc/porting_guide.md]: typo: stepos -> steps

File: doc/porting_guide.md [hunk 5]
```diff
@@ -3,268 +3,284 @@
...
 
 ## Supported Platforms
 
 The Xively Client has been deployed on many different devices already shipping in the IoT space running the following platform stacks:
 
-  - POSIX
-  - Microchip
+  - POSIX systems (OSX, Linux)
+  - Microchip TCP SDK
   - WMSDK
```
Comment (olgierdh) [inline doc/porting_guide.md]: This is used on Marvel's devices.

File: doc/porting_guide.md [hunk 6]
```diff
@@ -3,268 +3,284 @@
...
+The make system uses dash '-' separated string to define the TARGET and CONFIGuration for a build. 
+
+TARGET defines the build environment for specific platforms and toolchains.
+
+CONFIG flags specify selections from a list of Xively Client modules and features to be included in the Xively C Client library.
 
 _Default_ values will be chosen for you if you do not set the TARGET or CONFIG makefile variables.
 
-The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order and are separated by the - (minus sign / dash) character.
+The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order.  Each flag isseparated by the - (minus sign / dash) character.
```
Comment (atigyi) [inline doc/porting_guide.md]: isseperated is not separated

File: doc/porting_guide.md (hunk 4 above)
Comment (olgierdh) [inline doc/porting_guide.md]: stepos - typo - steps

File: doc/porting_guide.md [hunk 7]
```diff
@@ -3,268 +3,284 @@
...
-2. [Xively Client versions](#xively-client-versions)
+2. [C Client Overview](#c-client-overview)
 3. [Supported Platforms](#supported-platforms)
 4. [Board Support Package (BSP)](#board-support-package-bsp)
 5. [Porting the C Client](#porting-the-c-client)
-    - [Preceding Step - A Native Build](#preceding-step---a-native-build)
-    - [The Porting Step](#the-porting-step)
+    - [Suggested First Step - A Native Build](#suggested-first-step---a-native-build)
+    - [Porting the Xively C Client to Your Platform](#porting-the-xively-c-client-to-your-platform)
 6. [Xively Client Features](#xively-client-features)
```
Comment (DellaBitta) [inline doc/porting_guide.md]: noted!

File: doc/porting_guide.md (hunk 2 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: sure.

File: doc/porting_guide.md [hunk 8]
```diff
@@ -3,268 +3,284 @@
...
+    posix_io-posix_fs-thread_module-posix_platform-tls-senml-memory_limiter
 
 ###### Xively Client Feature flags
 
     - control_topic     - the control topic feature is in development.
                           Currently activating this feature has no behavioral benefit.
     - senml             - turns on SENML JSON serialization support for timeseries data. To maintain a small
                           footprint size we recommend turning this off (by removing senml flag from CONFIG).
-    - threading         - turns on threading: application callbacks will be called on separate thread.
+    - threading         - POSIX only.  This turns causes pplication callbacks to be called on separate thread.
```
Comment (atigyi) [inline doc/porting_guide.md]: ? This causes application callbacks to ... ?

File: doc/porting_guide.md [hunk 9]
```diff
@@ -3,268 +3,284 @@
...
+##### CONFIG Flags
 
 A typical CONFIG flag:
 
-    posix_io-posix_fs-thread_module-posix_platform-tls-senml-control_topic-memory_limiter
+    posix_io-posix_fs-thread_module-posix_platform-tls-senml-memory_limiter
 
 ###### Xively Client Feature flags
 
     - control_topic     - the control topic feature is in development.
```
Comment (atigyi) [inline doc/porting_guide.md]: shouldn't we remove this control topic description and any mention on it in this doc. I don't think it will be activated in a year.

File: doc/porting_guide.md [hunk 10]
```diff
@@ -3,268 +3,284 @@
...
 - cmake is needed by the integration test framework _cmocka_
 
-On OSX these are available through _brew_:
+On OSX these are available through `brew`:
 
     - brew install autoconf
     - brew install automake
     - brew install libtool
-    - brew install cmake
+    - brew install cmak
```
Comment (olgierdh) [inline doc/porting_guide.md]: typo - cmake

File: doc/porting_guide.md [hunk 11]
```diff
@@ -3,268 +3,284 @@
...
     - mqtt_localhost    - the Client will connect to localhost MQTT server. The purpose of this configuration option
                           is to help development processes.
     - no_certverify     - lowers security by disabling TLS certificate verification
-    - tls               - build with TLS module for securely connecting with TLS 1.2 to servers. The lack of this
-                          flag will skip the TLS handshaking which may helpful in connecting to mosquitto to test,
+    - tls               - the Client will use third party TLS 1.2 implementations to encrypt data before it's sent over network sockets.   The lack of this
+                          flag will skip the TLS handshaking.  This may helpful in connecting to mosquitto to test,
                           but may not be supported on some secure services and is not recommend for production
-                          deployment
+                          deployment.  The Xively Gateway will not accepts connections without TLS.
```
Comment (atigyi) [inline doc/porting_guide.md]: will not accept ? -s?

File: doc/porting_guide.md (hunk 8 above)
Comment (olgierdh) [inline doc/porting_guide.md]: typo - pplication - application

File: doc/porting_guide.md [hunk 12]
```diff
@@ -3,268 +3,284 @@
...
                           The memory monitor part is useful for memory consumption tracking and to hunt down memory
-                          leaks.
+                          leaks.  When a leak occurs a stack trace of the initial allocation will be logged.
     - mqtt_localhost    - the Client will connect to localhost MQTT server. The purpose of this configuration option
                           is to help development processes.
     - no_certverify     - lowers security by disabling TLS certificate verification
-    - tls               - build with TLS module for securely connecting with TLS 1.2 to servers. The lack of this
-                          flag will skip the TLS handshaking which may helpful in connecting to mosquitto to test,
+    - tls               - the Client will use third party TLS 1.2 implementations to encrypt data before it's sent over network sockets.   The lack of this
+                          flag will skip the TLS handshaking.  This may helpful in connecting to mosquitto to test,
```
Comment (olgierdh) [inline doc/porting_guide.md]: Shouldn't it be "This may be... " ?

File: doc/porting_guide.md (hunk 11 above)
Comment (olgierdh) [inline doc/porting_guide.md]: accepts ? or accept ?

File: doc/porting_guide.md [hunk 13]
```diff
@@ -354,30 +378,31 @@ Let's assume the new platform's name is np2000. And an early advise: as a rule o
...
+ 		- **networking** (`include/bsp/xi_bsp_io_net.h`)
+ 		- **memory** (`include/bsp/xi_bsp_mem.h`)
+ 		- **time** (`include/bsp/xi_bsp_time.h`)
+ 		- **random** (`include/bsp/xi_bsp_rng.h`)
+ 
+    Hint: to reach successful build just create the files and implement all the BSP API functions with **empty body**.  While this wont execute properly, it should link and run.
 
 - [x] select TLS implementation
-    - default is wolfssl, if wolfssl fits the needs then nothing to do here
+    - the default seelction is wolfssl.  If wolfssl fits the needs then nothing to do here.
```
Comment (olgierdh) [inline doc/porting_guide.md]: typo seelction

File: doc/porting_guide.md [hunk 14]
```diff
@@ -282,7 +298,7 @@ The release version flags without SENML and memory limits may look like this:
 
 For CI configurations please look at the file [.travis.yml](../../.travis.yml).
 
-#### Example Xively Client Applications linked with the library
+#### Example Xively Client Applications Linked with the Library
 
 Example application binaries can be found under directory _xi\_client\_c/src/examples_.
 These examples use of the Xively C Client library for connecting to Xively Servers,
```
Comment (atigyi) [inline doc/porting_guide.md]: use of?

File: doc/porting_guide.md [hunk 15]
```diff
@@ -297,49 +313,57 @@ The examples are command line applications libxively links against. They require
 
     - [ wolfssl | mbedtls ] - selects the TLS library which is used to provide secure connection with Xively servers, WolfSSL is set by default
 
-### The Porting Step
+### Porting the Xively C Client to Your Platform
+
+#### Let's Begin
+
+As a rule of thumb if you are stuck or confused, then please user existing platform config files like: `make/mt-os/mt-cc3200` or `make/mt-os/mt-linux` files as an example.
```
Comment (atigyi) [inline doc/porting_guide.md]: "please user existing" ?

File: doc/porting_guide.md [hunk 16]
```diff
@@ -297,49 +313,57 @@ The examples are command line applications libxively links against. They require
...
 
-#### Cross-compilation with the _make_ build system
+Let's assume the new platform's name is np2000. 
 
-To cross-compile the Xively C Client on OSX or Linux/Unix to a new fictitious platform *np2000* the following command should be enough
+
+#### The Goal: Cross Compilation with the _make_ Build System
+
+The ideal goal to cross-compile the Xively C Client on OSX or Linux/Unix to your platform.  For the sake of this tutorial, let's pretend that your platform is a new fictitious platform: The *NP2000*.  You'r platform sounds awesome.
```
Comment (atigyi) [inline doc/porting_guide.md]: you'r ?

File: doc/porting_guide.md (hunk 6 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: lol

File: doc/porting_guide.md (hunk 9 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: actually it's used by the file system updates.
so yes, I think within a year it will be used!

File: doc/porting_guide.md [hunk 17]
```diff
@@ -297,49 +313,57 @@ The examples are command line applications libxively links against. They require
...
+
+Our goal is construct the following command to build the Xively Client for the NP2000:
```
Comment (atigyi) [inline doc/porting_guide.md]: Our goal is to build the Xively C Client for the NP2000 with the following command:
?

File: doc/porting_guide.md [hunk 18]
```diff
@@ -354,30 +378,31 @@ Let's assume the new platform's name is np2000. And an early advise: as a rule o
                 XI_BSP_PLATFORM = np2000
                 XI_TARGET_PLATFORM = np2000
 
-- [x] extend *mt-os* to let the build system pick up the *mt-np2000* config file
+- [x] extend `make/mt-os/mt-os` to check the TARGET make parameter for 'np2000' and include the `make/mt-os/mt-np2000` config file when the np200 TARGET is found:
```
Comment (atigyi) [inline doc/porting_guide.md]: np2000 instead of np200 :)

File: doc/porting_guide.md (hunk 13 above)
Comment (atigyi) [inline doc/porting_guide.md]: selection

File: doc/porting_guide.md (hunk 17 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: no the goal is actually to build the command itself by altering the build system..  I will admit that it's an odd goal.

File: doc/porting_guide.md [hunk 19]
```diff
@@ -390,51 +415,53 @@ Let's assume the new platform's name is np2000. And an early advise: as a rule o
...
-    Modules from these 'incomplete' BSP implementations could be used as substitutes. For instance, on devices that mirror POSIX completely except for networking, like Simplelink, the networking module from src/bsp/platform/posix could be ovewritten with the one from src/bsp/platform/simplelink_incomplete. 
-- import one of the BSP TLS implementations in src/bsp/tls.  Currently we provide two different TLS BSP implementations: WolfSSL or mbedTLS.
-- add all of the directories in src/libxively to your include path
-- add the /include directory to your include path
+    Modules from these 'incomplete' BSP implementations could be used as substitutes. For instance, on devices that mirror POSIX completely except for networking, like Simplelink, the networking module from `src/bsp/platform/posix` could be ovewritten with the one from `src/bsp/platform/simplelink_incomplete`. 
+- import one of the BSP TLS implementations in `src/bsp/tls`.  Currently we provide two different TLS BSP implementations: WolfSSL or mbedTLS.
+- add all of the directories in `src/libxively` to your include path
```
Comment (atigyi) [inline doc/porting_guide.md]: this is already said in the 1st bulletpoint

File: doc/porting_guide.md (hunk 9 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: Additionally our default make includes control topics so we have to at least mention it.

File: doc/porting_guide.md (hunk 18 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: it's like 10x better than the np200!

Comment (DellaBitta) [inline doc/porting_guide.md]: fixed.

File: doc/porting_guide.md (hunk 19 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: first bullet point is to import the source files, this bullet point is to add them to the include path.

Comment (DellaBitta) [inline doc/porting_guide.md]: I clarified it a bit.

File: doc/porting_guide.md [hunk 20]
```diff
@@ -3,268 +3,292 @@
...
-#### Example make command
+The Platform Selector configurations will eventually be sunset.  Currently these configure the build system to include Critical Section implementations for invoking callbacks on new threads.  
+
+We suggest defining `posix_platform` and omit `threading` from your CONFIG options when building for custom platforms.  `threading` is currently ommitted by default.
+
+For more information about thread safe callback support please see the Xively C Client User Guide: `doc/user_guide.md`.
+
+#### Example make Command
 
 By executing a simple 'make' under directory xi\_client\_c should be sufficient on OSX or Linux/Unix. This will result in build configuration with the following default flags:
```
Comment (atigyi) [inline doc/porting_guide.md]: as I saw this is the last occurence of xi_client_c.
Maybe "... under root directory should be sufficient..." is ok, isn't it?

File: doc/porting_guide.md [hunk 21]
```diff
@@ -3,268 +3,292 @@
...
 
-If neither wolfSSL nor mbedTLS fits your target platform the build system is open for further BSP TLS implementations. But before selecting your target TLS solution be sure it supports:
+If neither wolfSSL nor mbedTLS fits your target platform the build system can be configured to use other BSP TLS implementations.
+
+Before selecting another TLS solution other than those mentioned above, be sure that it supports:
 
 - asynchronous networking
 - loading custom root CA certs from a buffer
-- OCSP
+- Online Certificate Status Protocol [(OCSP)](https://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol)
```
Comment (atigyi) [inline doc/porting_guide.md]: OCSP Stapling

File: doc/porting_guide.md [hunk 22]
```diff
@@ -3,268 +3,292 @@
...
-Invoking `make` without parameters will default to `make XI_BSP_PLATFORM=posix XI_BSP_TLS=wolfssl`. This is fine for building on OSX or Linux/Unix using wolfSSL for a TLS library, for cross-compilation you most likely will need to define both parameters explicitly. For instance:
+This is fine for building on OSX or Linux/Unix using wolfSSL for a TLS library.  However, for cross-compilation you will most likely need to define both parameters explicitly. For instance:
 
-`make XI_BSP_PLATFORM=cc3200 XI_BSP_TLS=microSSL`
+	`make XI_BSP_PLATFORM=cc3200 XI_BSP_TLS=microSSL`
 
-This command assumes directory *src/bsp/platform/cc3200* containing networking, memory, random and time implementations as well as directory *src/bsp/tls/microSSL* covering the TLS calls.
+*Note: microSSL is a fictional TLS library we've conjured-up as a quick custom example.*
+
+This command assumes directory `src/bsp/platform/cc3200*` containing networking, memory, random and time implementations as well as directory `src/bsp/tls/microSSL` covering the TLS calls.
```
Comment (atigyi) [inline doc/porting_guide.md]: asterisk remained at the end of the path

File: doc/porting_guide.md [hunk 23]
```diff
@@ -3,268 +3,292 @@
...
+    posix_io-posix_fs-posix_platform-tls-senml-memory_limiter
 
 ###### Xively Client Feature flags
 
     - control_topic     - the control topic feature is in development.
                           Currently activating this feature has no behavioral benefit.
     - senml             - turns on SENML JSON serialization support for timeseries data. To maintain a small
                           footprint size we recommend turning this off (by removing senml flag from CONFIG).
-    - threading         - turns on threading: application callbacks will be called on separate thread.
+    - threading         - POSIX only.  This turns causes application callbacks to be called on separate thread.
```
Comment (atigyi) [inline doc/porting_guide.md]: turns + causes, probably either is enough

File: doc/porting_guide.md [hunk 24]
```diff
@@ -3,268 +3,292 @@
...
-
     - expose_fs         - adds a new Xively C Client API function which allows external definition
-                          of File System calls
+                          of File System calls.  This will be part of our BSP system in the future.
 
 ###### Development flags
 
-    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration
-                          option is to aid development processes by simulating low-memory environment behavior.
+    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration is to aid development processes by simulating low-memory environment behavior.
```
Comment (atigyi) [inline doc/porting_guide.md]: unfortunately not manually breaking these lines ends up in needing the reader to scroll horizontally. Breaking manually brings more convenient experience for the reader, so pls break those manually as animals break lines. sorry.

File: doc/porting_guide.md (hunk 20 above)
Comment (atigyi) [inline doc/porting_guide.md]: additionally the default CONFIG and TARGET flags are outdated, pls update new ones are:

CONFIG:          [bsp_posix-posix_fs-posix_platform-tls-senml-control_topic-memory_limiter]
TARGET:          [osx-static-release]

File: doc/porting_guide.md [hunk 25]
```diff
@@ -354,30 +377,31 @@ Let's assume the new platform's name is np2000. And an early advise: as a rule o
...
+   	 - `src/bsp/np2000/xi_bsp_mem_np2000.c`
+   	 - `src/bsp/np2000/xi_bsp_rng_np2000.c`
+   	 - `src/bsp/np2000/xi_bsp_time_np2000.c`
+ 	- in these sourcefiles, define the functions declared in these corresponding Xively C Client BSP headers:
+ 		- **networking** (`include/bsp/xi_bsp_io_net.h`)
+ 		- **memory** (`include/bsp/xi_bsp_mem.h`)
+ 		- **time** (`include/bsp/xi_bsp_time.h`)
+ 		- **random** (`include/bsp/xi_bsp_rng.h`)
+ 
+    Hint: to reach successful build just create the files and implement all the BSP API functions with **empty body**.  While this wont execute properly, it should link and run.
```
Comment (atigyi) [inline doc/porting_guide.md]: + "You may use files under `src/bsp/platform/dummy` as empty starting point.

File: doc/porting_guide.md [hunk 26]
```diff
@@ -390,51 +414,57 @@ Let's assume the new platform's name is np2000. And an early advise: as a rule o
...
-    - you have to also provide a script *xively-client-c/src/import/tls/download_and_compile_myTLSlibrary.sh* which downloads the source of the custom TLS library and builds it. As a sample to follow look at the two already existing solutions: *download_and_compile_wolfssl.sh* and *download_and_compile_mbedtls.sh* in the same directory.
+    - create a file `make/mt-config/mt-tls-myTLSlibrary.mk` and fill in with content similar to `make/mt-config/mt-tls-wolfssl.mk` or `make/mt-config/mt-tls-mbedtls.mk`. This is where you would defined the include directory, the static library directory, the static library file to link against and the config flags of the custom TLS library.
+    - you might also want to provide a script `src/import/tls/download_and_compile_myTLSlibrary.sh` which downloads the source of the custom TLS library and builds it. As a sample to follow look at the two already existing solutions: `download_and_compile_wolfssl.sh`and `download_and_compile_mbedtls.sh` in that same directory.
```
Comment (atigyi) [inline doc/porting_guide.md]: As a sample to follow look at the...
Isn't follow OR look enough?

Again you haven't touched this part, so sorry, just recognised this may be a confusing spot.

File: doc/porting_guide.md (hunk 20 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: done.

File: doc/porting_guide.md (hunk 26 above)
Comment (DellaBitta) [inline doc/porting_guide.md]: honestly this isn't a porting step.  Why would a customer ever do this for themselves?  I'm going to remove it.

File: doc/porting_guide.md [hunk 27]
```diff
@@ -3,343 +3,366 @@
...
 
-`cd examples`
+	cd examples
+	make
 
-`make`
+Here if the library was previously built with the mbedTLS setting then the XI_BSP_TLS variable is necessary.  Otherwise you will get link errors.
 
-here if previously library was built with mbedTLS setting the XI_BSP_TLS variable is necessary to tell the linker which TLS lib to link agains:
+	make XI_BSP_TLS=mbedTLS
```
Comment (atigyi) [inline doc/porting_guide.md]: unfortunately here all lower case is required since this will be inserted into the template file name: mt-tls-$(XI_BSP_TLS).mk . And the filename is stilll lowercase :(

**Comment by atigyi** [comment]
one final note there about changing the letters in XI_BSP_TLS=mbedtls -> mbedTLS . This would need the file renames as well :(
Otherwise perfect and super precise job!! 💯
